core/templates/pages/exploration-editor-page/translation-tab/state-translation/state-translation.component.ts

```diff
@@ -189,15 +205,39 @@ angular.module('oppia').component('stateTranslation', {
           );
           activeContentId = activeContent.contentId;
           if (activeContent instanceof SubtitledUnicode) {
-            activeDataFormat = WRITTEN_TRANSLATION_TYPE_UNICODE;
+            activeDataFormat = TRANSLATION_DATA_FORMAT_UNICODE;
+          }
+        } else if (tabId === $scope.TAB_ID_RULE_INPUTS) {
+          if ($scope.interactionRuleTranslatableContents.length === 0) {
+            throw new Error(
+              'Accessed rule input translation tab when there are no rules');
           }
+
+          // Note that only 'TextInput' and 'SetInput' have translatable rule
+          // types. The rules tab is disabled for other interactions.
+          const {
+            rule, inputName, contentId
+          } = $scope.interactionRuleTranslatableContents[0];
+          activeContentId = contentId;
+          const inputType = rule.inputTypes[inputName];
+          activeDataFormat = RULE_INPUT_TYPES_TO_DATA_FORMATS[inputType];
+          $scope.activeRuleContentIndex = 0;
         }
         TranslationTabActiveContentIdService.setActiveContent(
-          activeContentId,
-          activeDataFormat);
+          activeContentId, activeDataFormat);
         $scope.activatedTabId = tabId;
       };
 
+      $scope.getHumanReadableRuleInputValues = function(inputValue, inputType) {
+        if (inputType === 'TranslatableSetOfNormalizedString') {
+          return ('[' + inputValue.normalizedStrSet.join(', ') + ']');
+        } else if (inputType === 'TranslatableSetOfUnicodeString') {
+          return ('[' + inputValue.unicodeStrSet.join(', ') + ']');
```

Comment:
Just curious, why not just use template literals instead of concatinating strings using '+'?

Reply:
No strong reason. I think we've done the latter in the codebase typically so I did it here too.